internal/world/azure.go

```diff
@@ -0,0 +1,207 @@
+package world
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+const (
+	AzureSubscriptionId    string = "AZURE_SUBSCRIPTION_ID"
+	AzureClientId          string = "AZURE_CLIENT_ID"
+	AzureClientSecret      string = "AZURE_CLIENT_SECRET"
+	AzureKeyVaultUrl       string = "AZURE_KEY_VAULT_URL"
+	AzureApiVersion        string = "AZURE_API_VERSION"
+	VaultUrl               string = "https://vault.azure.net"
+	MicrosoftLoginUrl      string = "https://login.microsoftonline.com/"
+	ClientCredentialsGrant string = "client_credentials"
+)
+
+type Oauth2Res struct {
```

Comment:
Please prefix all types that are only relevant for Azure functionality with `Azure` :)